react-redux/react-redux.d.ts

```diff
@@ -14,51 +14,72 @@ declare namespace ReactRedux {
   type Dispatch<S> = Redux.Dispatch<S>;
   type ActionCreator<A> = Redux.ActionCreator<A>;
 
-  interface ComponentDecorator<TOriginalProps, TOwnProps> {
-    (component: ComponentClass<TOriginalProps>|StatelessComponent<TOriginalProps>): ComponentClass<TOwnProps>;
+
+  interface ComponentConstructor<P, S> extends __React.ComponentLifecycle<P, S> {
```

Comment:
What purpose does this class serve? It looks a lot like `Component`, but slightly stricter in the constructor. Also, `S` in `ComponentLifecycle` is taken to be "state"; usage of it as `context` is not correct.